hub/core/storage/s3/s3.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+from hub.core.storage.provider import Provider
+from hub.core.storage.s3.s3_mapper import S3Mapper
+
+
+class S3Provider(Provider):
+    """
+    Provider class for using S3 storage.
+    """
+
+    def __init__(
+        self,
+        root: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        endpoint_url: Optional[str] = None,
+        aws_region: Optional[str] = None,
+        max_pool_connections: Optional[int] = 25,
+    ):
+        """
+        Initializes the S3Provider
+
+        Example:
+            s3_provider = S3Provider("snark-test/benchmarks")
+
+        Args:
+            root (str): the root of the provider.
```

Comment:
you should say something like "All read/write request keys will be appended to `root`".